### PR TITLE
[Feature] Add talent source selection

### LINKED ISSUE #17187: ✨ Add talent source selection to talent request creation
### Details

Add UI for selecting talent sources when creating a talent request, and display the selected sources on the admin detail view. This allows request creators to specify which talent sources should be used to find matching candidates.

Blocked by #17185.

Related: #17184, #17185, #17186

### Design file & designer

@Jerryescandon 

[Public facing: Find talent page](https://www.figma.com/design/9V8kBIWxOXRaN9dkcVCU1o/Manager---Talent-search--OLD-?node-id=1275-17853&t=DlO8S04lc0kIv3yf-1)
[Public facing: Request form ](https://www.figma.com/design/oUIFoHHAZ5FwFcGHc8p7Yt/Manager---Request-form?node-id=1103-9202&t=pgnqJGFF3zVy5xxI-1)
[Admin side: Request details](https://www.figma.com/design/OcJlqTARk7UEG2aK7H0AG5/Talent-requests--Communities-?node-id=146-18641&t=1vgBpZK6UR1FBj7J-1)

### 📸 Screenshots

<details><summary>📸 ON "FIND TALENT" FORM SELECTION</summary>
<p>

Image

</p>
</details> 

<details><summary>📸 ON "FIND TALENT" RESULT CARDS </summary>
<p>

Image

</p>
</details> 

<details><summary>📸 ON "SUMMARY OF FILTERS" FORM </summary>
<p>

Image

</p>
</details> 

<details><summary>📸 ON ADMIN SIDE "REQUEST DETAILS" </summary>
<p>

Image

</p>
</details> 

### 🌎 Localization

- Form field labels and option text need English and French translations

### Proposed Implementation

- Add a talent source selection field to the request creation form at `apps/web/src/pages/SearchRequests/RequestPage/components/RequestForm.tsx` (line ~241)
- The `createTalentRequest` mutation (schema.graphql line ~3800) may need to be updated to accept and persist the selected sources
- Update the admin-side talent request detail view to display the selected talent sources

### ✅ Acceptance Criteria

- [ ] In 'Find talent' page: Add new "Talent source" checkbox
- [ ] In 'Find talent' page: Update copy on group and level selections
- [ ] In 'Find talent' page: Update cards in results section
- [ ] In 'Request form' page: Update "summary of filters" info to include talent source selections
- [ ] For Dev: Selected talent sources are persisted via the `createTalentRequest` mutation
- [ ] In admin 'Request details" displays the selected talent sources
- [ ] Form field labels available in both English and French

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #17187 

## 👋 Introduction

- Adds checklist input on Find Talent page to select one or more talent sources
- Adds the talent sources to the "Summary of filters" section of the Request page in a check boolean icon list
- Adds the talent sources to the "Source of talent" section of the Details tab when viewing a talent request
- Adds new result cards for communities

## 🧪 Testing

1. Build app `make refresh`
2. Login as admin
3. Go to Find talent page. 
4. Go through flow of creating a talent request:
  - Talent source checklist works correctly
  - Filters are added to the `applicantFilter` in the `createTalentRequest` mutation
  - Summary of filters sets the boolean icons correctly
  - Source of talent section has the correct talent source filters, and icons are correct (on admin side) 

## 📸 Screenshot

<details><summary>Find talent page</summary>
<p>

image

</p>
</details> 

<details><summary>Summary of filters (Request form page)</summary>
<p>

image

</p>
</details> 

<details><summary>Source of talent card</summary>
<p>

image

</p>
</details>

### CODE REVIEW

**Comment by yonikid15** [comment]
I can't re-recreate the playwright error locally using dev or CI setup. @mauli0110 do you have any idea's as to why the pool card visibility keeps failing on CI?

Screenshot from failed checks:
image

**Comment by mauli0110** [comment]
> I can't re-recreate the playwright error locally using dev or CI setup. @mauli0110 do you have any idea's as to why the pool card visibility keeps failing on CI?
> 
> Screenshot from failed checks: image

Yeah, I had this issue before some weeks but then it got resolved by its own, let me check on your branch

File: api/app/GraphQL/Queries/CountTalentRequestMatchesByCommunity.php
Comment (esizer) [inline api/app/GraphQL/Queries/CountTalentRequestMatchesByCommunity.php:1]: Hmm, we were talking yesterday about these queries being prety expensive to run and I'm wondering if we can try and take another pass at this one.

This seems to create an n+3 since we map over all return values and run 3 separate user queries on each row returned. 

Unfortunatley, I don't have many good suggestions here outside of maybe trying to see if you are able to model it after the existing pool queries this replaces? 🤔

File: apps/web/src/pages/SearchRequests/SearchPage/components/CommunityResultCard.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        value={community.id}
+        onClick={() => {
+          setValue("pool", "");
+          setValue("communityId", community.id);
+          setValue("count", count);
+        }}
+      >
+        {intl.formatMessage(
+          {
+            defaultMessage: "Request all matching candidates {count}",
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/components/CommunityResultCard.tsx:111]: @Jerryescandon Did we want to put the number in parentheses? I dont feel stronfgly about this but it does look a little odd here.

Image

File: apps/web/src/pages/SearchRequests/SearchPage/components/FormFields.tsx [hunk 2]
```diff
@@ -150,20 +160,80 @@ const FormFields = ({
...
+          />
+          <p>
+            {intl.formatMessage({
+              defaultMessage:
+                "What is the intended classification and work stream of this position?",
+              id: "VpolrX",
+              description:
+                "Question above classification and work stream filter",
+            })}
+          </p>
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/components/FormFields.tsx:236]: At a minnimum, I beleive this should be associated witht he input via an id and `aria-describedby`. But, I'm also wondering if we want to use the input context prop which does thta for us just with some styles applied.

If the styles are not wanted then the aria attribute is just as good.

@Jerryescandon do you have any thoughts or opinions?

Image
Image

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx [hunk 3]
```diff
@@ -171,62 +201,59 @@ export const SearchForm = ({
...
-                    description:
-                      "Heading for total matching candidates across a certain number of pools in results section of search page.",
-                  },
-                  {
-                    totalCandidateCount: candidateCount,
-                    numPools: results?.length ?? 0,
-                  },
-                )}
+                {intl.formatMessage({
+                  defaultMessage: "Results:",
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx]: I think the preference here is to use `commonMessages.dividingColon` reather than embedding them in strings.

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx [hunk 4]
```diff
@@ -171,62 +201,59 @@ export const SearchForm = ({
...
+                          count={communityCount}
+                          showQualifiedInPool={showQualifiedInPool}
+                          showAtLevel={showAtLevel}
+                        />
+                      ),
+                    )}
                   </div>
+                  {results?.length ? (
+                    <>
+                      <p className="mt-9 mb-1.5 text-2xl lg:text-3xl">
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx]: I think this should be a heading component. You can apply the required overrides to match styles if that was the reason for not using.

File: apps/web/src/pages/SearchRequests/SearchPage/hooks.ts [hunk 5]
```diff
@@ -72,6 +83,12 @@ interface UseCandidateCountReturn {
     count: number;
     pool: SearchResultCard_PoolFragment;
   }[];
+  communities: {
+    community: { id: string; name?: { localized?: string | null } | null };
+    qualifiedInPoolCount: number;
+    atLevelCount: number;
+    count: number;
+  }[];
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/hooks.ts]: This is a nit pick so feel free to ingore since I see you were just following the established pattern in this file but, inline types like this are discouaged in typescript due to overuse of them causing some slight performance issues.

Its not a huge deal but, if we can, we should try and avoid them.

**Review by esizer — DISMISSED** [review]
Awesome! I have a few changes hopefully most are not too diffiuclkt. The difficult one will be the query itself 😓

**Review by tristan-orourke — COMMENTED** [review]
Code review: 3 finding(s) posted as inline comments.

---
🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>If this review was useful, react with 👍. Otherwise, react with 👎.</sub>

File: apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx [hunk 6]
```diff
@@ -92,6 +110,28 @@ const TalentRequestSourcesCard = ({ query }: TalentRequestSourcesCardProps) => {
...
+        >
+          <Ul unStyled noIndent inside>
+            {talentSourceOptionsFiltered.map((source) => (
+              <li key={source.value}>
+                <BoolCheckIcon
+                  value={selectedTalentSources.includes(source.value)}
+                  trueLabel={intl.formatMessage(commonMessages.selected)}
+                  falseLabel={intl.formatMessage(commonMessages.notSelected)}
+                >
+                  {source.value === TalentRequestSource.QualifiedInPool &&
```
Comment (tristan-orourke) [inline apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx]: The `BoolCheckIcon` children only handle `QualifiedInPool` and `AtLevel` explicitly. If a new `TalentRequestSource` enum value is added beyond `Advancement`, this will render a `BoolCheckIcon` with no label text. Consider adding a fallback to `source.label?.localized` for unrecognized values. The same pattern appears in `SearchRequestFilters.tsx:170` and `FormFields.tsx` (the checklist builder has its own fallback already).

---
<sub>🤖 Generated with [Claude Code](https://claude.ai/code)</sub>

File: apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx [hunk 7]
```diff
@@ -140,6 +154,31 @@ const ApplicantFilters = ({
...
+                {talentSourceOptionsFiltered.map((source) => (
+                  <li key={source.value}>
+                    <BoolCheckIcon
+                      value={filterTalentSources.includes(source.value)}
+                      trueLabel={intl.formatMessage(commonMessages.selected)}
+                      falseLabel={intl.formatMessage(
+                        commonMessages.notSelected,
+                      )}
+                    >
+                      {source.value === TalentRequestSource.QualifiedInPool &&
```
Comment (tristan-orourke) [inline apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx]: Same empty-label issue as in `TalentRequestSourcesCard` — no fallback for unrecognized `TalentRequestSource` values. A future enum addition would render a blank `BoolCheckIcon`. Adding a fallback branch using `source.label?.localized` would cover it.

---
<sub>🤖 Generated with [Claude Code](https://claude.ai/code)</sub>

File: api/tests/Feature/TalentRequestMatchesTest.php [hunk 8]
```diff
@@ -581,6 +615,80 @@ public function testCountByPoolCountsDistinctUsersAndExcludesNonMatchingPools():
...
+                ],
+            ],
+        ])
+            ->assertJsonPath('data.countTalentRequestMatchesByCommunity.0.community.id', $community->id)
+            ->assertJsonPath('data.countTalentRequestMatchesByCommunity.0.qualifiedInPoolCount', 2)
+            ->assertJsonPath('data.countTalentRequestMatchesByCommunity.0.atLevelCount', 2)
+            ->assertJsonPath('data.countTalentRequestMatchesByCommunity.0.count', 3);
+    }
+
+    public function testCountByCommunityExcludesCommunitiesWithNoMatches(): void
```
Comment (tristan-orourke) [inline api/tests/Feature/TalentRequestMatchesTest.php:702]: Good test coverage for the explicit talent source selections. Worth adding a test case for when `talentSources` is empty/omitted — the backend treats that as "all sources selected" via `TalentRequestSource::selected()`, and it would be good to pin that default behavior.

---
<sub>🤖 Generated with [Claude Code](https://claude.ai/code)</sub>

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx (hunk 3 above)
Comment (yonikid15) [reply apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx]: [Use dividing colon](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/695bd8750f10175dad139935115b7fc5429a0b8d)

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx (hunk 4 above)
Comment (yonikid15) [reply apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx]: [Change to heading](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/ba071607b09751ca603ab71ec898384139b0c807)

File: apps/web/src/pages/SearchRequests/SearchPage/hooks.ts (hunk 5 above)
Comment (yonikid15) [reply apps/web/src/pages/SearchRequests/SearchPage/hooks.ts]: [Fix type](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/f63de4b4f14fc590f74c266b0b0ce116ec32708e)

File: api/app/GraphQL/Queries/CountTalentRequestMatchesByCommunity.php
Comment (yonikid15) [reply api/app/GraphQL/Queries/CountTalentRequestMatchesByCommunity.php:1]: [Improve query](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/f22b514b0abef6442952fe8b94bf36dd81a82a3e)

### The fix — push the grouping into SQL

1. Build two subqueries, each still going through the full
   `whereMatchesTalentRequest` filtering, just selecting bare columns instead
   of aggregating:
   ```sql
   SELECT user_id, community_id, 'pool' AS source
   FROM pool_candidates
   JOIN pools ON pools.id = pool_candidates.pool_id
   -- ...whereMatchesTalentRequest filters...
   ```
   ```sql
   SELECT user_id, community_id, 'interest' AS source
   FROM community_interests
   -- ...whereMatchesTalentRequest filters...
   ```

2. Combine them with `UNION ALL` into one result set of
   `(user_id, community_id, source)` rows.

3. Wrap that as a derived table (`fromSub`) and run **one** `GROUP BY community_id`
   with conditional aggregation:
   ```sql
   SELECT community_id,
     COUNT(DISTINCT CASE WHEN source = 'pool' THEN user_id END) AS qualified_in_pool_count,
     COUNT(DISTINCT CASE WHEN source = 'interest' THEN user_id END) AS at_level_count,
     COUNT(DISTINCT user_id) AS count
   FROM (pool_rows UNION ALL interest_rows) AS matches
   GROUP BY community_id
   ```
   The `COUNT(DISTINCT CASE WHEN ...)` pattern gets us all three numbers —
   including the deduplicated union total — in one pass, since
   `COUNT(DISTINCT user_id)` across the whole group naturally collapses a
   user who appears via both `pool` and `interest` rows for the same
   community into one.

4. Batch-fetch the `Community` models in one `whereIn('id', ...)` instead of
   `find()`-ing per row.

### Result

Query count goes from `2 + 4N` (scales with the number of matching
communities) to a flat **2**, regardless of N — matching the same
single-`GROUP BY`-query shape that `countTalentRequestMatchesByPool` already
used, which is what motivated the whole rewrite.

File: apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx (hunk 6 above)
Comment (yonikid15) [reply apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx]: Nice catch! Added fallback message in [Add fallback message](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/940b42fd31099d9d0d3fdc28800990f9eff8a23e)

File: apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx (hunk 7 above)
Comment (yonikid15) [reply apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx]: Nice catch! Added fallback message in [Add fallback message](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/940b42fd31099d9d0d3fdc28800990f9eff8a23e)

File: api/tests/Feature/TalentRequestMatchesTest.php (hunk 8 above)
Comment (yonikid15) [reply api/tests/Feature/TalentRequestMatchesTest.php:702]: Sounds good! Added another test for omitted sources in [Improve testing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/17399/commits/53407d91ccb6c64359af43cbd79d9ca5d2863d68)

File: apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx [hunk 9]
```diff
@@ -158,24 +158,36 @@ const ApplicantFilters = ({
...
+                    intl.formatMessage(commonMessages.notAvailable);
+                  if (source.value === TalentRequestSource.QualifiedInPool) {
+                    label = intl.formatMessage(
+                      talentRequestMessages.qualifiedInPoolLabel,
+                    );
+                  } else if (source.value === TalentRequestSource.AtLevel) {
+                    label = intl.formatMessage(
+                      talentRequestMessages.atLevelLabel,
+                    );
+                  }
```
Comment (tristan-orourke) [inline apps/web/src/components/SearchRequestFilters/SearchRequestFilters.tsx]: It's a small, purely stylistic change, but this seems like a good use for a map object:
```suggestion
                  const talentSourceLabels = {
                    [TalentRequestSource.QualifiedInPool]: talentRequestMessages.qualifiedInPoolLabel,
                    [TalentRequestSource.AtLevel]: talentRequestMessages.atLevelLabel,
                  };
                  const label = talentSourceLabelOverrides[source.value]
                    ? intl.formatMessage(talentSourceLabelOverrides[source.value])
                    : (source.label?.localized ?? intl.formatMessage(commonMessages.notAvailable));
```

File: apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx [hunk 10]
```diff
@@ -114,22 +114,30 @@ const TalentRequestSourcesCard = ({
...
-                  value={selectedTalentSources.includes(source.value)}
-                  trueLabel={intl.formatMessage(commonMessages.selected)}
-                  falseLabel={intl.formatMessage(commonMessages.notSelected)}
-                >
-                  {source.value === TalentRequestSource.QualifiedInPool &&
-                    intl.formatMessage(
-                      talentRequestMessages.qualifiedInPoolLabel,
+            {talentSourceOptionsFiltered.map((source) => {
+              let label = source.label?.localized ?? notProvided;
+              if (source.value === TalentRequestSource.QualifiedInPool) {
```
Comment (tristan-orourke) [inline apps/web/src/pages/TalentRequests/components/TalentRequestSourcesCard.tsx]: Same here.

**Review by tristan-orourke — APPROVED** [review]
Looks great Yoni! I left one tiny suggestion but even that is subjective. 

I discovered one bug while testing but it's unrelated to any changes in this PR specifically, so I'll make another ticket for simplicity.

File: apps/web/src/pages/SearchRequests/SearchPage/components/CommunityResultCard.tsx (hunk 1 above)
Comment (yonikid15) [reply apps/web/src/pages/SearchRequests/SearchPage/components/CommunityResultCard.tsx:111]: Discussed: This will be added to a separate ticket https://github.com/GCTC-NTGC/gc-digital-talent/issues/17467

File: apps/web/src/pages/SearchRequests/SearchPage/components/FormFields.tsx (hunk 2 above)
Comment (yonikid15) [reply apps/web/src/pages/SearchRequests/SearchPage/components/FormFields.tsx:236]: Discussed: This will be added to a separate ticket https://github.com/GCTC-NTGC/gc-digital-talent/issues/17467
